src/main/java/net/barrage/school/java/ecatalog/app/services/ProductSyncServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package net.barrage.school.java.ecatalog.app.services;
+
+import jakarta.transaction.Transactional;
+import net.barrage.school.java.ecatalog.app.product_sources.ProductSource;
+import net.barrage.school.java.ecatalog.model.Merchant;
+import net.barrage.school.java.ecatalog.repository.MerchantRepository;
+import net.barrage.school.java.ecatalog.repository.ProductRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.StreamSupport;
+
+@Service
+public class ProductSyncServiceImpl implements ProductSyncService{
+
+    @Autowired
+    private ProductRepository productRepository;
+    @Autowired
+    private MerchantRepository merchantRepository;
+    private final List<ProductSource> productSources;
+    public ProductSyncServiceImpl(
+            List<ProductSource> productSources) {
+        this.productSources = productSources;
+    }
+
+    @Transactional
+    @Override
+    public void seedDatabase(){
+        for (ProductSource source : productSources) {
+            syncWithProductSource(source);
+        }
+    }
+    @Transactional
+    @Scheduled(cron = "0 0 * * * *") //Every hour
+    void syncDatabase() {
+        for (ProductSource source : productSources) {
+            if(source.isRemote()) {
+                syncWithProductSource(source);
+            }
+        }
+    }
+    @Transactional
+    public void syncWithMerchantName(String name) {
+        for (ProductSource source : productSources) {
+            if(source.getMerchantName().equalsIgnoreCase(name)) {
+                syncWithProductSource(source);
+            }
+        }
+    }
+
+    @Transactional
+    private void syncWithProductSource(ProductSource source) {
+        Merchant merchant = merchantRepository.findByName(source.getMerchantName());
+        if(merchant  != null) merchant.getProducts().clear();
```

Comment:
- Pls avoid using `if` without `{}`. It's kind of tradition)
- Have u tried this `merchant.getProducts().clear()`? Will try myself, but it looks wrong) Usually u can delete hibernate entity only by calling `delete*` explicitly.

Reply:
I tried it and it worked, but i can change it to delete explicitly if it is more clear

Reply:
Strange, but it works) Okay)